.github/workflows/codeql.yml

```diff
@@ -0,0 +1,66 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "main" ]
+  schedule:
+    - cron: '18 6 * * 5'
```

Comment:
Minor: Could you please add a helpful comment that "this will run At 06:18 on Fridays" (probably will help folks not familiar with cron expressions)

Reply:
Thanks for the suggestion @nkthakur48. Added the comment.